pallets/capacity/src/lib.rs

```diff
@@ -477,6 +715,36 @@ impl<T: Config> Pallet<T> {
 		Ok(capacity)
 	}
 
+	fn increase_stake_and_issue_boost_capacity(
+		staker: &T::AccountId,
+		staking_details: &mut StakingDetails<T>,
+		target: &MessageSourceId,
+		amount: &BalanceOf<T>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		staking_details.deposit(*amount).ok_or(ArithmeticError::Overflow)?;
+		Self::set_staking_account_and_lock(staker, staking_details)?;
+
+		// get the capacity generated by a Provider Boost
+		let capacity = Self::capacity_generated(T::RewardsProvider::capacity_boost(*amount));
+
+		let mut target_details = Self::get_target_for(staker, target).unwrap_or_default();
+
+		target_details.deposit(*amount, capacity).ok_or(ArithmeticError::Overflow)?;
+		Self::set_target_details_for(staker, *target, target_details);
```

Comment:
I also was not expecting set_target_details to also delete details perhaps a more descriptive name might be nice here such as `set_or_delete_target_details`.

(minor) Also, a more idiomatic way to rewrite the function in rust
```
fn update_or_delete_target_details(
    staker: &T::AccountId,
    target: MessageSourceId,
    target_details: StakingTargetDetails<BalanceOf<T>>,
) {
    match target_details.amount.is_zero() {
        true => StakingTargetLedger::<T>::remove(staker, target),
        false => StakingTargetLedger::<T>::insert(staker, target, target_details),
    }
}
```